.gitlab-ci.yml

```diff
@@ -123,18 +129,30 @@ test:debug:
   image: $CI_IMAGE_DOCKER
   stage: test
   timeout: 1h
+  cache:
+    key: $CI_PIPELINE_ID
+    paths:
+      - cache/caches/
+      - cache/notifications/
+    policy: pull
   script:
     - git fetch --depth=1 origin master
     - rm -rf ~/.gradle/daemon/
     - CODECOV_TOKEN=$(aws ssm get-parameter --region us-east-1 --name ci.dd-sdk-android.codecov-token  --with-decryption --query "Parameter.Value" --out text)
-    - GRADLE_OPTS="-Xmx2560m" ./gradlew :unitTestDebug --stacktrace --no-daemon
+    - GRADLE_OPTS="-Xmx2560m" ./gradlew :unitTestDebug --stacktrace --no-daemon --build-cache --gradle-user-home cache/
     - bash <(cat ./codecov.sh) -t $CODECOV_TOKEN
 
 test:release:
   tags: [ "runner:main" ]
   image: $CI_IMAGE_DOCKER
   stage: test
   timeout: 1h
+  cache:
```

Comment:
shouldn't you specify to use the cache here : `./gradlew :unitTestRelease --build-cache` or I did not get how the caching works ?

Reply:
good catch! indeed I forgot to do it.